src/main/java/duke/core/Duke.java

```diff
@@ -0,0 +1,339 @@
+package duke.core;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.databse.Database;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+public class Duke {
+
+    private Database database;
+    private ArrayList<Task> taskList;
+    private UI ui = new UI();
+
+    /**
+     * Constructor
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        database = new Database(filePath);
+    }
+
+
+    /**
+     * search for a task
+     * @param tasklist
+     * @param keyword
+     * @return
+     */
+    public ArrayList<Task> searchTask(ArrayList<Task> tasklist, String keyword) {
+        ArrayList<Task> result = new ArrayList<>();
+        for (Task task: tasklist) {
+            if (task.searchKeyword(keyword)) {
+                result.add(task);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * start the program
```

Comment:
Should this start with a word 'Starts' instead of 'start'? I have noticed similar points in your javadoc elsewhere as well.

Reply:
I see! Thanks.